PusherClient/Pusher.cs

```diff
@@ -100,24 +101,14 @@ public Pusher(string applicationKey, PusherOptions options = null)
 
         #region Public Methods
 
+        [MethodImpl(MethodImplOptions.Synchronized)]

```

Comment:
I've updated the code.

From what I can tell, I don't think this is require on Disconnect.